sw/device/lib/dif/dif_plic.h

```diff
@@ -0,0 +1,271 @@
+  kDifPlicFlashCtrlRdLvl = 50,          /**< Flash read FIFO filled to lvl */
+  kDifPlicFlashCtrlOpDone = 51,         /**< Flash operation complete */
+  kDifPlicFlashCtrlOpError = 52,        /**< Flash operation failed with err */
+  kDifPlicHmacDone = 53,                /**< HMAC done */
+  kDifPlicHmacFifoFull = 54,            /**< HMAC FIFO full */
+  kDifPlicHmacErr = 55,                 /**< HMAC error */
+  kDifPlicAlertHandlerClassA = 56,      /**< Alert Handler class A alert */
+  kDifPlicAlertHandlerClassB = 57,      /**< Alert Handler class B alert */
+  kDifPlicAlertHandlerClassC = 58,      /**< Alert Handler class C alert */
+  kDifPlicAlertHandlerClassD = 59,      /**< Alert Handler class D alert */
+  kDifPlicNmiGenEsc0 = 60,              /**< NMI Gen escalation interrupt 0 */
+  kDifPlicNmiGenEsc1 = 61,              /**< NMI Gen escalation interrupt 1 */
+  kDifPlicNmiGenEsc2 = 62,              /**< NMI Gen escalation interrupt 2 */
+  kDifPlicNmiGenEsc3 = 63,              /**< NMI Gen escalation interrupt 3 */
+  kDifPlicLastIrq = kDifPlicNmiGenEsc3, /**< The last valid IRQ ID */
+} dif_plic_irq_src_id_t;
+
+/**
+ * PLIC IRQ source peripheral enumeration
+ *
+ * Enumeration used to determine which peripheral asserted the corresponding
+ * IRQ source.
+ */
+typedef enum dif_plic_irq_peripheral {
+  kDifPlicPeripheralGpio = 0,     /**< GPIO */
+  kDifPlicPeripheralUart,         /**< UART */
+  kDifPlicPeripheralSpiDevice,    /**< SPI */
+  kDifPlicPeripheralFlashCtrl,    /**< Flash */
+  kDifPlicPeripheralHmac,         /**< HMAC */
+  kDifPlicPeripheralAlertHandler, /**< Alert handler */
+  kDifPlicPeripheralNmiGen,       /**< NMI generator */
+  kDifPlicPeripheralCount,        /**< Number of PLIC peripherals */
+} dif_plic_irq_peripheral_t;
+
+/**
+ * PLIC external interrupt target
+ *
+ * Enumeration used to determine which set of IE, CC, threshold registers to
+ * access dependent on the target.
+ */
+typedef enum dif_plic_target {
+  kDifPlicTargetIbex0 = 0, /* Ibex CPU */
+  kDifPlicTargetCount,
+} dif_plic_target_t;
+
+/**
+ * Generic enable/disable enumeration
+ *
+ * Enumeration used to enable/disable bits, flags, ...
+ */
+typedef enum dif_plic_enable {
+  kDifPlicEnable = 0, /**< enable */
+  kDifPlicDisable,    /**< disable */
+} dif_plic_enable_t;
+
+/**
+ * Generic set/unset enumeration
+ *
+ * Enumeration used to set/unset, or get the state of bits,flags ...
+ */
+typedef enum dif_plic_flag {
+  kDifPlicSet = 0, /**< set */
+  kDifPlicUnset,   /**< unset */
+} dif_plic_flag_t;
+
+/**
+ * PLIC CC (IRQ claim register data)
+ *
+ * Data that is populated by the PLIC DIF when interrupt is claimed. Is used by
+ * the PLIC DIF consumers to establish which peripheral issued an IRQ, and the
+ * type of this IRQ.
+ */
+typedef struct dif_irq_claim_data {
+  dif_plic_irq_peripheral_t peripheral; /**< Peripheral that interrupted */
+  dif_plic_irq_src_id_t source;         /**< IRQ type */
```

Comment:
You write `type` in the comment, but the member is named `source`. You might want to update one of the two.

Reply:
done